src/main/java/org/apache/mesos/hdfs/config/SchedulerConf.java

```diff
@@ -1,10 +1,28 @@
 package org.apache.mesos.hdfs.config;
 
+import com.google.inject.Singleton;
 import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.conf.Configured;
+import org.apache.hadoop.fs.Path;
 
-public class SchedulerConf extends MainConf {
-  public SchedulerConf(Configuration conf, int configServerPort) {
-    super(conf, configServerPort);
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Properties;
+
+@Singleton
+public class SchedulerConf extends Configured {
+
+  public SchedulerConf(Configuration conf) {
+    setConf(conf);
+  }
+
+  public SchedulerConf() {
+    Properties props = System.getProperties();
```

Comment:
Hm, I see your point it is nice to give users flexibility to configure this, but I'd rather keep the code simple if possible. I think we could require users to change etc/hadoop/mesos-site.xml since it is already included as part of the framework. Is there any reason why this would be problematic for them?